pkg/cloudprovider/drift.go

```diff
@@ -73,3 +84,33 @@ func (c *CloudProvider) isAMIDrifted(ctx context.Context, machine *v1alpha5.Mach
 	}
 	return !lo.Contains(lo.Keys(mappedAMIs), instance.ImageID), nil
 }
+
+// Checks if the security groups are drifted, by comparing the AWSNodeTemplate.Status.SecurityGroups
+// to the ec2 instance security groups
+func (c *CloudProvider) areSecurityGroupsDrifted(ec2Instance *instance.Instance, nodeTemplate *v1alpha1.AWSNodeTemplate) (bool, error) {
+	// nodeTemplate.Spec.SecurityGroupSelector can be nil if the user is using a launchTemplateName to define SecurityGroups
+	// Karpenter will not drift on changes to securitygroup in the launchTemplateName
+	if nodeTemplate.Spec.LaunchTemplateName != nil {
+		return false, nil
+	}
+
+	securityGroupIds := sets.New(lo.Map(nodeTemplate.Status.SecurityGroups, func(sg v1alpha1.SecurityGroup, _ int) string { return sg.ID })...)
+	if len(securityGroupIds) == 0 {
+		return false, fmt.Errorf("no security groups exist given constraints")
```

Comment:
This is the same error message as in the AWSNodeTemplate controller, right? We're not actually computing constraints here, just querying from the status. Can it be something like the following?
```suggestion
		return false, fmt.Errorf("no security groups exist in the AWSNodeTemplate Status")
```